src/vcpkg/registries.cpp

```diff
@@ -1135,14 +1146,21 @@ namespace vcpkg
         return r.array_elements(arr, underlying);
     }
 
-    LockFile::Entry LockFile::get_or_fetch(const VcpkgPaths& paths, StringView key)
+    LockFile::Entry LockFile::get_or_fetch(const VcpkgPaths& paths, StringView repo, StringView reference)
     {
-        auto it = lockdata.find(key);
+        auto lockdata_key = repo.to_string();
+        if (reference != "HEAD")
+        {
+            // old-style, only repo. implicitly HEAD
+            Strings::append(lockdata_key, "@", reference);
+        }
+
+        auto it = lockdata.find(lockdata_key);
```

Comment:
```suggestion
        auto it = lockdata.find(lockdata_key);
        if (it == lockdata.end() && reference == "HEAD")
        {
            it = lockdata.find(StringView(lockdata_key).substr(0, lockdata_key.size() - 5));
        }
```

Reply:
I don't think this is necessary -- in fact to minimize the possibility of collisions with usernames, I think we should not have this fallback.